src/screens/datasharing/components/BaseTekUploadView.tsx

```diff
@@ -71,12 +71,16 @@ export const BaseTekUploadView = ({
   };
   const handleUpload = useCallback(async () => {
     setLoading(true);
+    isUploading(true);
+
     try {
       await fetchAndSubmitKeys(contagiousDateInfo);
       setLoading(false);
+      isUploading(false);
```

Comment:
I would find this a bit easier to read if the function was named `setIsUploading(false)`